create_superuser.sh

```diff
@@ -0,0 +1,4 @@
+echo "from django.contrib.auth.models import User; \
+      User.objects.filter(email='admin@example.com').delete(); \
+      User.objects.create_superuser('admin', 'admin@example.com', 'admin')" \
+  | python manage.py shell
```

Comment:
Varför inte använda `python manage.py createsupercuser`? Verkar lite lättare.

Reply:
Det går bra det med, men då måste en köra `docker exec -it 981698615 bash` och sedan leka runt i containern :) På detta sätt kan en snabbt skriva `docker exec -it 981698615 bash ./create_superuser.sh`. Så egentligen behövs det inte, men jag tyckte det var tydligare så här för nya användare.

Reply:
Varför funkar inte `docker exec -it python manage.py <command>` ? Det är oroväckande eftersom typ alla django grejer sköts på det sättet.

Reply:
Det fungerar, men python manage.py createsupercuser kräver en interaktiv användare för att mata in ett lösenord. Finns ingen flagga för att sätta det. Men på detta sätt så kan vi speca ett lösenord också :)

Reply:
Det finns flaggor...

Reply:
Yes, men inte för lösenord :(
```
  --username USERNAME   Specifies the login for the superuser.
  --noinput, --no-input
                        Tells Django to NOT prompt the user for input of any
                        kind. You must use --username with --noinput, along
                        with an option for any other required field.
                        Superusers created with --noinput will not be able to
                        log in until they're given a valid password.
```

Reply:
Om det finns en `--password` flagga som inte är dokumenterad så blir jag positivt överraskad :)

Reply:
Se edit ovan. Also no password flag is so lame, god damn it Django.

Reply:
Eller så skapar vi det här shell kommandot. Är det för att det känns för hackigt som gör att det känns fel?

Reply:
Yes, pumpa in script i ett "interaktivt" skal känns inte alls bra. Dessutom är det förvirrande för en som kan lite Django att det finns ett till createsuperuser kommando.

Fixtures låter dig ha mer saker, vilket ju alltid är nice. Det är i princip en databas dump man laddar. Så man kan ha Södran redan inprogrammerad, med alla stolar, ett gäng exempel biljetter, blogposter etc

Reply:
Då låter fixtures som en kanon grej! Men inte något jag hinner/kan kasta ihop. 
Så mitt förslag är att vi behåller det så här just nu och så skulle vi kunna kolla på fixtures när någon av oss har tid.

btw, så behöver verkligen inte skalet en öppnar mot containern vara interaktivt. Målet här är just att skippa interaktiviteten och skapa ett kommando som du kan köra lokalt istället :P

Reply:
Det är på just den sidan jag hittade det skript som vi nu diskuterar ^^ Men tycker inte `python3 manage.py -c "samma pythonkod fast efteråt"` är så mycket bättre :P

Reply:
Nu tänkte jag sova :) Men det jag tycker vi borde ha är ett kommando på en rad som skapar en admin på servern. Detta för att det smidigt ska kunna dokumenteras i `README.md`. I just denna lösning är jag inte så kär, har du ett eget förslag som åstadkommer samma mål får du gärna lägga upp det :)

Reply:
Det är precis vad loaddata gör. I princip går det till så här:

1. Sätt igång din app med en fräsch databas
2. Meka i admin, eller i detta fall skapa en användare via createsuperuser
3. `python manage.py dumpdata`
4. ???
5. PROFIT

Typ så, sedan kan det behövas ett par flaggor, men det förklarar Django bättre än mig.